pcs/registry.py

```diff
@@ -44,24 +55,35 @@
 class Registry(abc.ABC):
     @staticmethod
     def from_dict(input_dict: Dict) -> "Registry":
-        return InMemoryRegistry(input_dict=input_dict)
+        return InMemoryRegistry.from_dict(input_dict)
 
     @staticmethod
     def from_yaml(file_path: str) -> "Registry":
         with open(file_path) as file_in:
             config = yaml.safe_load(file_in)
-        reg = InMemoryRegistry(input_dict=config)
-        updated_specs = []
-        for spec_id, spec_body in reg.specs.items():
-            if spec_body["type"] == "LocalRepo":
-                p = PurePath(spec_body["path"])
-                if not p.is_absolute():
-                    abs_path = (Path(file_path).parent / p).resolve()
-                    updated_spec = Spec.from_flat(copy.deepcopy(spec_body))
-                    updated_spec.update_body({"path": str(abs_path)})
-                    updated_specs.append((spec_id, updated_spec))
-        for old_id, updated in updated_specs:
-            reg.replace_spec(old_id, updated)
+        reg = InMemoryRegistry.from_dict(config)
+        reg._make_relative_local_repo_paths_absolute(Path(file_path).parent)
+        return reg
+
+    @staticmethod
+    def from_yamls(file_paths: List[str]) -> "Registry":
+        SPECS_KEY = InMemoryRegistry.SPECS_KEY
+        ALIASES_KEY = InMemoryRegistry.ALIASES_KEY
+        input_dict = {SPECS_KEY: {}, ALIASES_KEY: {}}
+        parent_path = Path(file_paths[0]).parent
+        for path in file_paths:
+            # TODO(andyk): Remove this constraint.
```

Comment:
Is this TODO still valid?

Reply:
Unfortunately, yes. We don't currently support loading registry files from different directories. If you want to understand why, I can explain more in a voice or video chat.

I had been exploring the idea of supporting `imports` to address (really, it would be more like getting around it), since we've been talking about that for along time, but that is a non-trivial feature to add.